app/src/main/java/com/portfolio/tasky/entry/fragments/LoginFragment.kt

```diff
@@ -141,18 +140,30 @@ class LoginFragment : Fragment(), FragmentInflater by FragmentInflaterImpl(), Te
             )
         )
     }
-    /**
-     * Necessary for navigating to RegistrationFragment.
-     * Currently not called
-     */
+
+    private fun startLogin() {
+        val email = viewBinding.etEmail.subLayout.etInput.text.toString()
+        val password = viewBinding.etPassword.subLayout.etInput.text.toString()
+
+        lifecycleScope.launch(Dispatchers.IO + coroutineExceptionHandler){
+            viewModel.login(AuthenticationRequest(email, password))
+        }
```

Comment:
I don't know how your `login()` function of the ViewModel looks like, but since you execute it in a suspend function, I strongly assume it's a suspend function. Never expose suspend functions from a ViewModel to the UI. If you press login here and then a config change happens (such as a screen rotation), your activity will be destroyed, `lifecycleScope` will be cancelled and logging in as well.
Instead, always launch coroutines in `viewModelScope` and update state afterwards which you observe in the UI 👍 

You very rarely really need `lifecycleScope` in Android.

Reply:
Thanks for the information!

Done